src/pages/settings/Wallet/WalletPage/CardDetails.js

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+import * as Expensicons from '../../../../components/Icon/Expensicons';
+import MenuItemWithTopDescription from '../../../../components/MenuItemWithTopDescription';
+import Clipboard from '../../../../libs/Clipboard';
+import useLocalize from '../../../../hooks/useLocalize';
+import usePrivatePersonalDetails from '../../../../hooks/usePrivatePersonalDetails';
+import ONYXKEYS from '../../../../ONYXKEYS';
+import * as CartUtils from '../../../../libs/CardUtils';
+
+const propTypes = {
+    /** Card number */
+    pan: PropTypes.string,
+
+    /** Card expiration date */
+    expiration: PropTypes.string,
+
+    /** 3 digit code */
+    cvv: PropTypes.string,
+
+    /** User's private personal details */
+    privatePersonalDetails: PropTypes.shape({
+        /** User's home address */
+        address: PropTypes.shape({
+            street: PropTypes.string,
+            city: PropTypes.string,
+            state: PropTypes.string,
+            zip: PropTypes.string,
+            country: PropTypes.string,
+        }),
+    }),
+};
+
+const defaultProps = {
+    pan: '',
+    expiration: '',
+    cvv: '',
+    privatePersonalDetails: {
+        address: {
+            street: '',
+            street2: '',
+            city: '',
+            state: '',
+            zip: '',
+            country: '',
+        },
+    },
+};
+
+function CardDetails({pan, expiration, cvv, privatePersonalDetails}) {
+    usePrivatePersonalDetails();
+    const {translate} = useLocalize();
+
+    const handleCopyToClipboard = () => {
```

Comment:
Unfortunately it won't work for this UI, because `CopyTextToClipboard` displays a text component that we don't need as it's already displayed by `MenuItemWithTopDescription`. However, I found out that this would work: 

```javascript
<MenuItemWithTopDescription
    ...
    shouldShowRightComponent
    rightComponent={
        <PressableWithDelayToggle
            tooltipText={translate('reportActionContextMenu.copyToClipboard')}
            tooltipTextChecked={translate('reportActionContextMenu.copied')}
            icon={Expensicons.Copy}
            onPress={handleCopyToClipboard}
        />
    }
    ...
/>
```

So basically, a part of `CopyTextToClipboard` that displays the icon, what do you think?

Reply:
Awesome, let's do that instead!

Reply:
Done 😄